docs/shrc_diag_util.md

```diff
@@ -262,106 +282,168 @@ Line number information for classes in the cache is then shown with one of the f
 
-#### `Maximum space for AOT bytes`
- : The maximum number of bytes of AOT-compiled code that can be stored in the cache.
+#### Class paths
 
-#### `JIT data bytes`
-: The number of bytes of JIT-related data stored in the cache.
+The following example shows one class path with 4 entries:
 
-#### `Reserved space for JIT data bytes`
-: The number of bytes reserved for JIT-related data in the cache.
+        1: 0x2234FA6C CLASSPATH
+                /myVM/Apps/application1.jar
+                /myVM/Apps/application2.jar
+                /myVM/Apps/application3.jar
+                /myVM/Apps/application4.jar
 
-#### `Maximum space for JIT data bytes`
-: The maximum number of bytes of JIT-related data that can be stored in the cache.
+- `1`: the ID of the VM that wrote this data.
+- `0x2234FA6C`: the address where this data is stored.
+- `CLASSPATH`: the type of data that was written.
 
-#### `Zip cache bytes`
-: The number of zip entry cache bytes stored in the cache.
+#### ROMClasses
 
-#### `Startup hint bytes`
-: The number of bytes of data stored to describe startup hints.
+This example shows an entry for a single `ROMClass`:
 
-#### `Data bytes`
-: The number of bytes of non-class data stored by the VM.
+        1: 0x2234F7DC ROMCLASS: java/lang/Runnable at 0x213684A8
+                Index 1 in class path 0x2234FA6C
 
-#### `Metadata bytes`
-: The number of bytes of data stored to describe the cached classes. :fontawesome-solid-pencil-alt:{: .note aria-hidden="true"} **Note:** This field is available only in the top layer cache output or when a cache is not layered.
+- `1`: the ID of the VM that wrote this data.
+- `0x2234F7DC`: the address where the metadata about the class is stored.
+- `ROMCLASS`: the type of data that was stored.
+- `java/lang/Runnable`: the name of the class.
+- `0x213684A8`: the address where the class was stored.
+- `Index 1`: the index in the class path where the class was loaded from.
+- `0x2234FA6C`: the address of the class path against which this class is stored.
 
-#### `Metadata % used`
-: The proportion of metadata bytes to class bytes, which indicates how efficiently cache space is being used. The value shown does consider the `Class debug area size`.
+Stale classes are marked with `!STALE!`. Any partition or modification context that is used when the class is stored is also shown.
 
-#### `Class debug area size`
-: The size in bytes of the Class Debug Area. This area is reserved to store `LineNumberTable` and `LocalVariableTable` class attribute information.
+#### AOT methods
 
-#### `Class debug area bytes used`
-: The size in bytes of the Class Debug Area that contains data.
+This example shows an entry for one AOT-compiled method:
 
-#### `Class debug area % used`
-: The percentage of the Class Debug Area that contains data.
+        1: 0x540FBA6A AOT: loadConvert
+                for ROMClass java/util/Properties at 0x52345174
```

Comment:
This example can be updated:

```
1: 0x00007F841A800554 AOT: hashCode Signature: ()I Address: 0x00007F83F6859280
        for ROMClass java/lang/Object at 0x00007F83F6859000.
```

We now print out the signature and the address of the rom method.
In the example, signature is `()I`, rom method address is `0x00007F83F6859280`.

Reply:
Please check the explanations I used for this example to make sure I correctly transposed the parts.

Reply:
Looks good to me.